src/PIL/pdfParser.py

```diff
@@ -0,0 +1,814 @@
+import codecs
+import collections
+import io
+import mmap
+import os
+import re
+import sys
+import zlib
+
+try:
+    from UserDict import UserDict
+except ImportError:
+    UserDict = collections.UserDict
```

Comment:
Please could you comment which of these is for Python 2 and which for Python 3? It'll make it easier when we drop support for 2.7.